cpp/demo/poisson_matrix_free/CMakeLists.txt

```diff
@@ -0,0 +1,42 @@
+# This file was generated by running
+#
+#     python cmake/scripts/generate-cmakefiles from dolfinx/cpp
+#
+cmake_minimum_required(VERSION 3.16)
+
+set(PROJECT_NAME demo_poisson_matrix_free)
+project(${PROJECT_NAME} LANGUAGES C CXX)
+
+# Set C++17 standard
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)
+
+if (NOT TARGET dolfinx)
+  find_package(DOLFINX REQUIRED)
+endif()
+
+# Add target to compile UFL files
+if (PETSC_SCALAR_COMPLEX EQUAL 1)
+  set(SCALAR_TYPE "--scalar_type=double _Complex")
+endif()
+add_custom_command(
+  OUTPUT poisson.c
+  COMMAND ffcx ${CMAKE_CURRENT_SOURCE_DIR}/poisson.ufl ${SCALAR_TYPE}
+  VERBATIM DEPENDS poisson.ufl COMMENT "Compile poisson.ufl using FFCx")
+
+add_executable(${PROJECT_NAME} main.cpp ${CMAKE_CURRENT_BINARY_DIR}/poisson.c)
+target_link_libraries(${PROJECT_NAME} dolfinx)
+
+# Do not throw error for 'multi-line comments' (these are typical in
+# rst which includes LaTeX)
+include(CheckCXXCompilerFlag)
+CHECK_CXX_COMPILER_FLAG("-Wno-comment" HAVE_NO_MULTLINE)
+set_source_files_properties(main.cpp PROPERTIES COMPILE_FLAGS "$<$<BOOL:${HAVE_NO_MULTLINE}>:-Wno-comment -Wall -Wextra -pedantic -Werror>")
+
+# Test targets (used by DOLFINx testing system)
+set(TEST_PARAMETERS2 -np 2 ${MPIEXEC_PARAMS} "./${PROJECT_NAME}")
+set(TEST_PARAMETERS3 -np 3 ${MPIEXEC_PARAMS} "./${PROJECT_NAME}")
```

Comment:
MPI size 3 tests are very slow on GitHub runners

Reply:
At the moment we only call mpi_3  on circle ci. 
On Github actions we have:
```
  cd build/demo
  ctest -V -R demo -R serial
  ctest -V -R demo -R mpi_2
```